cli/pcluster/configure/easyconfig.py

```diff
@@ -219,14 +228,25 @@ def _create_vpc_parameters(vpc_section, scheduler, min_subnet_size, automate_vpc
 
 
 def _ask_for_subnets(subnet_list, vpc_section):
+    default_master_subnet = vpc_section.get_param_value("master_subnet_id")
     master_subnet_id = prompt_iterable(
-        "Master Subnet ID", subnet_list, default_value=vpc_section.get_param_value("master_subnet_id")
+        "Master Subnet ID",
+        subnet_list,
+        default_value=default_master_subnet
+        if default_master_subnet in [subnet_entry[0] for subnet_entry in subnet_list]
+        else None,
     )
 
+    default_compute_subnet = vpc_section.get_param_value("compute_subnet_id")
     compute_subnet_id = prompt_iterable(
         "Compute Subnet ID",
         subnet_list,
-        default_value=vpc_section.get_param_value("compute_subnet_id") or master_subnet_id,
+        default_value=(
+            default_compute_subnet
+            if default_compute_subnet in [subnet_entry[0] for subnet_entry in subnet_list]
```

Comment:
what about defining a `available_subnets= [subnet_entry[0] for subnet_entry in subnet_list]`
given that it is used in two places inside the _ask_for_subnets function?